### DISCUSSION TITLE
Mismatched python version for pip & python in PATH

### DISCUSSION THREAD

**Discussion by rgoldberg** [Homebrew/discussions#3452] [Everyday usage] [answered]
I've installed both `python@3.9` & `python@3.10`.

I update my packages (including `pip` & `setuptools`) for both versions.

I don't know how this happened, but now:

```zsh
$ python --version
Python 3.9.13

$ pip --version
pip 22.2 from /usr/local/lib/python3.10/site-packages/pip (python 3.10)
```

`python` is version 3.9.13 in my `PATH`, but `pip` is for python 3.10 in my `PATH`.

What would cause this mismatch? How can I prevent it from recurring?

**Comment by gromgit** [comment, marked as answer]
What's the output of:
```
which python pip
ls -ld $(which python pip)
```
